2009-06-13	Magnus Granberg		<zorry@gentoo.org>
		
		b.g.o #232601
		* gcc/varasm.c (decl_tls_model): Check flag_pic instead of flag_shlib.

--- gcc/varasm.c	2009-03-17 21:18:21.000000000 +0100
+++ gcc/varasm.c	2009-04-29 03:10:09.000000000 +0200
@@ -5607,7 +5607,11 @@
   bool is_local;
 
   is_local = targetm.binds_local_p (decl);
-  if (!flag_shlib)
+  #ifdef ENABLE_ESP
+    if (!flag_pic)
+  #else
+    if (!flag_shlib)
+  #endif
     {
       if (is_local)
 	kind = TLS_MODEL_LOCAL_EXEC;
